src/Tools/CsvImporter/Program.cs

```diff
@@ -1,251 +1,116 @@
-}
+string configFilePath = Path.Combine(configDirPath, "config.json");
+string logFilePath = Path.Combine(logsDirPath, $"CsvImporter_{DateTime.UtcNow:yyyy-MM-dd_HH-mm-ss}.log");
 
-// Read config file.
-CsvImporterConfig csvImporterConfig;
-try
-{
-    csvImporterConfig = await ConfigFileUtils.GetCsvImporterConfigAsync();
-}
-catch (Exception ex)
-{
-    throw new Exception($"Error reading config file: {ex.Message}");
-}
+bool configFileExists = File.Exists(configFilePath);
 
-QueueClientService queueClientService;
-try
+CsvImporterConfig? preLaunchConfig = null;
+if (configFileExists)
 {
-    string storageConnectionString = Environment.GetEnvironmentVariable("STORAGE_CONNECTION_STRING") ?? throw new NullReferenceException("STORAGE_CONNECTION_STRING environment variable not set");
-
-    queueClientService = new(
-        connectionString: storageConnectionString
+    preLaunchConfig = JsonSerializer.Deserialize(
+        json: File.ReadAllText(configFilePath),
+        jsonTypeInfo: CoreJsonContext.Default.CsvImporterConfig
     );
 }
-catch (NullReferenceException)
-{
-    ConsoleUtils.WriteWarning("STORAGE_CONNECTION_STRING environment variable not set.");
-    ConsoleUtils.WriteInfo("Attempting to use an Azure token credential instead.");
 
-    string queueUriString = csvImporterConfig.QueueUri ?? Environment.GetEnvironmentVariable("QUEUE_URI") ?? throw new NullReferenceException("QUEUE_URI environment variable not set or missing from config file.");
+var builder = Host.CreateApplicationBuilder(args);
+
+builder.Configuration
+    .AddEnvironmentVariables();
+
+builder.Logging
+    .ClearProviders()
+    .AddSimpleConsole(options =>
+    {
+        options.SingleLine = false;
+        options.UseUtcTimestamp = true;
+    })
+    .AddFileLogger(options =>
+    {
+        options.FilePath = logFilePath;
+    });
 
-    TokenCredential tokenCredential;
+if (preLaunchConfig is not null)
+{
     try
     {
-        tokenCredential = new ChainedTokenCredential(
-            new AzureCliCredential(),
-            new AzurePowerShellCredential(),
-            new ManagedIdentityCredential()
-        );
-
-        queueClientService = new(
-            queueUri: new(queueUriString),
-            tokenCredential: tokenCredential
-        );
+        builder.Services
+            .AddCsvImporterQueueClientService(
+                connectionString: AuthUtils.GetStorageConnectionString()
+            );
     }
-    catch (Exception ex)
+    catch (Exception)
     {
-        ConsoleUtils.WriteError($"Error getting access token for Azure: {ex.Message}");
-        return 22;
+        try
+        {
+            builder.Services
+                .AddCsvImporterQueueClientService(
+                    queueUri: Environment.GetEnvironmentVariable("QUEUE_URI") ?? preLaunchConfig.QueueUri ?? throw new NullReferenceException("QUEUE_URI environment variable not set or missing from config file."),
```

Comment:
Should probably drop using `QUEUE_URI` from an environment variable and just rely on what's in the config file.